src/Employer/Employer.Web/Controllers/EmployerVacancyController.cs

```diff
@@ -0,0 +1,67 @@
+using System.Threading.Tasks;
+using Esfa.Recruit.Employer.Web.Configuration;
+using Esfa.Recruit.Employer.Web.Configuration.Routing;
+using Esfa.Recruit.Employer.Web.Orchestrators;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Esfa.Recruit.Employer.Web.Controllers
+{
+    [Route(RoutePaths.AccountRoutePath)]
+    public class EmployerVacancyController : Controller
+    {
+        private readonly EmployerVacancyOrchestrator _employerVacancyOrchestrator;
+
+        public EmployerVacancyController(EmployerVacancyOrchestrator employerVacancyOrchestrator)
+        {
+            _employerVacancyOrchestrator = employerVacancyOrchestrator;
+        }
+
+        [HttpGet("employer-create-vacancy", Name = RouteNames.EmployerCreateVacancy_Get)]        
+        public async Task<IActionResult> CreateVacancy([FromQuery]string ukprn, [FromQuery]string programmeId)
+        {
+            TempData[TempDataKeys.ReferredFromMa] = true;
+
+            if(string.IsNullOrWhiteSpace(ukprn) && string.IsNullOrWhiteSpace(programmeId))
```

Comment:
Can you get rid of this check and just get _employerVacancyOrchestrator.GetProviderUkprn(ukprn) and _employerVacancyOrchestrator.GetProgrammeId(programmeId); to return immediately if values are null. That means we only have the simple IF statement at the bottom to work out the returned route.

Also by doing this it will remove the bug you have by coming from the MA Home page after you have come from the Favourite homepage as it doesn't clear the cookies.